--- a/include/efi/efidef.h	2024-07-29 10:20:45
+++ b/include/efi/efidef.h	2024-09-09 11:38:04
@@ -147,6 +147,7 @@ typedef enum {
     EfiMemoryMappedIO,
     EfiMemoryMappedIOPortSpace,
     EfiPalCode,
+    EfiPersistentMemory,
     EfiMaxMemoryType
 } EFI_MEMORY_TYPE;

--- a/arch/arm/efi/efi-boot.h	2024-07-29 10:20:45
+++ b/arch/arm/efi/efi-boot.h	2024-09-09 11:40:36
@@ -194,10 +194,11 @@ static EFI_STATUS __init efi_process_memory_map_bootin
              (desc_ptr->Type == EfiConventionalMemory ||
               desc_ptr->Type == EfiLoaderCode ||
               desc_ptr->Type == EfiLoaderData ||
-              (!map_bs &&
-               (desc_ptr->Type == EfiBootServicesCode ||
-                desc_ptr->Type == EfiBootServicesData))) )
-        {
+              desc_ptr->Type == EfiACPIReclaimMemory ||
+	      desc_ptr->Type == EfiPersistentMemory ||
+	      desc_ptr->Type == EfiBootServicesCode ||
+	       desc_ptr->Type == EfiBootServicesData) )
+      	      {
             if ( !meminfo_add_bank(bootinfo_get_mem(), desc_ptr) )
             {
                 PrintStr(L"Warning: All " __stringify(NR_MEM_BANKS) 
