diff --git a/arch/arm/cpuerrata.c b/arch/arm/cpuerrata.c
index 0248893de0..9349797009 100644
--- a/arch/arm/cpuerrata.c
+++ b/arch/arm/cpuerrata.c
@@ -185,8 +185,6 @@ warn:
     if ( !warned )
     {
         ASSERT(system_state < SYS_STATE_active);
-        warning_add("No support for ARM_SMCCC_ARCH_WORKAROUND_1.\n"
-                    "Please update your firmware.\n");
         warned = false;
     }
 
diff --git a/drivers/char/console.c b/drivers/char/console.c
index 913ae1b66a..b3e2c72a0c 100644
--- a/drivers/char/console.c
+++ b/drivers/char/console.c
@@ -56,12 +56,6 @@ string_runtime_param("conswitch", opt_conswitch);
 /* sync_console: force synchronous console output (useful for debugging). */
 static bool_t __initdata opt_sync_console;
 boolean_param("sync_console", opt_sync_console);
-static const char __initconst warning_sync_console[] =
-    "WARNING: CONSOLE OUTPUT IS SYNCHRONOUS\n"
-    "This option is intended to aid debugging of Xen by ensuring\n"
-    "that all output is synchronously delivered on the serial line.\n"
-    "However it can introduce SIGNIFICANT latencies and affect\n"
-    "timekeeping. It is NOT recommended for production use!\n";

 /* console_to_ring: send guest (incl. dom 0) console data to console ring. */
 static bool_t __read_mostly opt_console_to_ring;
@@ -945,7 +945,6 @@ void __init console_init_preirq(void)
         serial_start_sync(sercon_handle);
         add_taint(TAINT_SYNC_CONSOLE);
         printk("Console output is synchronous.\n");
-        warning_add(warning_sync_console);
     }
 }
 
