diff --git a/arch/arm/domain_build.c b/arch/arm/domain_build.c
index fb8fbb1650..3a92b39ce6 100644
--- a/arch/arm/domain_build.c
+++ b/arch/arm/domain_build.c
@@ -117,6 +117,14 @@ unsigned int __init dom0_max_vcpus(void)
     return opt_dom0_max_vcpus;
 }
 
+
+static unsigned long __initdata smbios_addr;
+void __init dmi_efi_get_table(const void *smbios, const void *smbios3)
+{
+	 smbios_addr = (unsigned long)smbios;
+}
+
+
 /*
  * Insert the given pages into a memory bank, banks are ordered by address.
  *
@@ -1101,6 +1109,14 @@ int __init make_hypervisor_node(struct domain *d,
     if ( res )
         return res;
 
+     if ( smbios_addr != 0 )
+     {
+	     res = fdt_property_u64(fdt, "xen,smbios", smbios_addr);
+	      if ( res )
+		       return res;
+     }
+
+
     /* Cannot use fdt_property_string due to embedded nulls */
     res = fdt_property(fdt, "compatible", compat, sizeof(compat));
     if ( res )
@@ -2043,6 +2059,14 @@ int __init construct_hwdom(struct kernel_info *kinfo,
     if ( rc < 0 )
         return rc;
 
+    if ( smbios_addr != 0 )
+    {
+	  map_regions_p2mt(d,
+			   gaddr_to_gfn(smbios_addr), 1,
+			     maddr_to_mfn(smbios_addr), p2m_mmio_direct_c);
+    }
+
+
     return construct_domain(d, kinfo);
 }
 
diff --git a/common/efi/boot.c b/common/efi/boot.c
index 2047d58e39..50349a826e 100644
--- a/common/efi/boot.c
+++ b/common/efi/boot.c
@@ -1131,12 +1131,10 @@ static void __init efi_tables(void)
             esrt = (UINTN)efi_ct[i].VendorTable;
     }
 
-#ifndef CONFIG_ARM /* TODO - disabled until implemented on ARM */
     dmi_efi_get_table(efi.smbios != EFI_INVALID_TABLE_ADDR
                       ? (void *)(long)efi.smbios : NULL,
                       efi.smbios3 != EFI_INVALID_TABLE_ADDR
                       ? (void *)(long)efi.smbios3 : NULL);
-#endif
 }
 
 static void __init setup_efi_pci(void)
