From d3b8b8369c5d48d88ea9ec024cf2988833e1ccaa Mon Sep 17 00:00:00 2001
From: Petr Fedchenkov <giggsoff@gmail.com>
Date: Tue, 16 Aug 2022 15:53:35 +0300
Subject: [PATCH] disable Werror to build under gcc 11.2

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>
---
 tools/libs/libs.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/libs/libs.mk b/tools/libs/libs.mk
index 2d973ccb95..fe85eeef22 100644
--- a/tools/libs/libs.mk
+++ b/tools/libs/libs.mk
@@ -11,7 +11,7 @@ MINOR ?= 0
 
 SHLIB_LDFLAGS += -Wl,--version-script=libxen$(LIBNAME).map
 
-CFLAGS   += -Werror -Wmissing-prototypes
+CFLAGS   += -Wmissing-prototypes
 CFLAGS   += -I./include $(CFLAGS_xeninclude)
 CFLAGS   += $(foreach lib, $(USELIBS_$(LIBNAME)), $(CFLAGS_libxen$(lib)))
 
-- 
2.34.1

