diff -r -u ../orig/dnsmasq-2.78/src/rfc2131.c src/rfc2131.c
--- a/src/rfc2131.c 2017-10-02 18:43:51.000000000 +0530
+++ b/src/rfc2131.c	2020-04-09 15:31:15.000000000 +0530
@@ -1026,11 +1026,10 @@
 		  int len;
 		  unsigned char *mac = extended_hwaddr(ltmp->hwaddr_type, ltmp->hwaddr_len,
 						       ltmp->hwaddr, ltmp->clid_len, ltmp->clid, &len);
-		  my_syslog(MS_DHCP | LOG_WARNING, _("not using configured address %s because it is leased to %s"),
+		  my_syslog(MS_DHCP | LOG_WARNING, _("pruning configured address %s because it is leased to %s"),
 			    addrs, print_mac(daemon->namebuff, mac, len));
+		  lease_prune(ltmp, now);
 		}
-	      else
-		{
 		  struct dhcp_context *tmp;
 		  for (tmp = context; tmp; tmp = tmp->current)
 		    if (context->router.s_addr == config->addr.s_addr)
@@ -1042,7 +1041,6 @@
 		    my_syslog(MS_DHCP | LOG_WARNING, _("not using configured address %s because it was previously declined"), addrs);
 		  else
 		    conf = config->addr;
-		}
 	    }
 	  
 	  if (conf.s_addr)
