diff --git a/src/rfc2131.c b/src/rfc2131.c
index c54c255..dad70df 100644
--- a/src/rfc2131.c
+++ b/src/rfc2131.c
@@ -1125,11 +1125,10 @@ size_t dhcp_reply(struct dhcp_context *context, char *iface_name, int int_index,
 		  int len;
 		  unsigned char *mac = extended_hwaddr(ltmp->hwaddr_type, ltmp->hwaddr_len,
 						       ltmp->hwaddr, ltmp->clid_len, ltmp->clid, &len);
-		  my_syslog(MS_DHCP | LOG_WARNING, _("not using configured address %s because it is leased to %s"),
-			    daemon->addrbuff, print_mac(daemon->namebuff, mac, len));
+		  my_syslog(MS_DHCP | LOG_WARNING, _("pruning configured address %s because it is leased to %s"),
+			   daemon->addrbuff, print_mac(daemon->namebuff, mac, len));
+		  lease_prune(ltmp, now);
 		}
-	      else
-		{
 		  struct dhcp_context *tmp;
 		  for (tmp = context; tmp; tmp = tmp->current)
 		    if (context->router.s_addr == config->addr.s_addr)
@@ -1141,7 +1140,6 @@ size_t dhcp_reply(struct dhcp_context *context, char *iface_name, int int_index,
		    my_syslog(MS_DHCP | LOG_WARNING, _("not using configured address %s because it was previously declined"), daemon->addrbuff);
 		  else
 		    conf = config->addr;
-		}
 	    }
 	  
 	  if (conf.s_addr)
--
2.49.0
