diff -r -u ../dnsmasq-orig/dnsmasq-2.75/src/option.c src/option.c
--- a/src/option.c	2015-07-30 12:59:07.000000000 -0700
+++ b/src/option.c	2017-06-13 15:14:58.109166818 -0700
@@ -2734,9 +2734,6 @@
 		  ret_err(_("prefix length must be exactly 64 for subnet constructors"));
 	      }
 
-	    if (new->prefix < 64)
-	      ret_err(_("prefix length must be at least 64"));
-	    
 	    if (!is_same_net6(&new->start6, &new->end6, new->prefix))
 	      ret_err(_("inconsistent DHCPv6 range"));
 
