From 0957b86d55dd37f4acc7135ea423ee7b614ac23d Mon Sep 17 00:00:00 2001
From: Petr Fedchenkov <giggsoff@gmail.com>
Date: Tue, 16 Aug 2022 16:18:29 +0300
Subject: [PATCH] disable Werror to build under gcc 11.2

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d4a7014f59..b487847b1f 100644
--- a/Makefile
+++ b/Makefile
@@ -207,7 +207,7 @@ CFLAGS += -fomit-frame-pointer
 endif
 
 CFLAGS += -nostdinc -fno-builtin -fno-common
-CFLAGS += -Werror -Wredundant-decls -Wno-pointer-arith
+CFLAGS += -Wredundant-decls -Wno-pointer-arith
 $(call cc-option-add,CFLAGS,CC,-Wvla)
 CFLAGS += -pipe -D__XEN__ -include $(BASEDIR)/include/xen/config.h
 CFLAGS-$(CONFIG_DEBUG_INFO) += -g
-- 
2.34.1

