From 3f1491ea2bca27dfa967ed7ad2f69fce6d15333d Mon Sep 17 00:00:00 2001
From: Daniel Lezcano <daniel.lezcano@linaro.org>
Date: Thu, 1 Jun 2017 11:28:49 +0200
Subject: [PATCH 17/45] cpuidle: Fix NULL driver checking

Signed-off-by: Daniel Lezcano <daniel.lezcano@linaro.org>
---
 drivers/cpuidle/driver.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/cpuidle/driver.c b/drivers/cpuidle/driver.c
index dc32f34e68d9..efbae090964e 100644
--- a/drivers/cpuidle/driver.c
+++ b/drivers/cpuidle/driver.c
@@ -274,9 +274,11 @@ EXPORT_SYMBOL_GPL(cpuidle_register_driver);
  */
 void cpuidle_unregister_driver(struct cpuidle_driver *drv)
 {
-	spin_lock(&cpuidle_driver_lock);
-	__cpuidle_unregister_driver(drv);
-	spin_unlock(&cpuidle_driver_lock);
+	if (drv) {
+		spin_lock(&cpuidle_driver_lock);
+		__cpuidle_unregister_driver(drv);
+		spin_unlock(&cpuidle_driver_lock);
+	}
 }
 EXPORT_SYMBOL_GPL(cpuidle_unregister_driver);
 
-- 
2.15.1

