From 3f3616c00f0e0fc68a7dafb98323c2a4ca88a024 Mon Sep 17 00:00:00 2001
From: Guodong Xu <guodong.xu@linaro.org>
Date: Tue, 23 May 2017 07:09:28 +0800
Subject: [PATCH 18/45] fix: bluetooth: hci_ll: set enable_gpio direction OUT
 at probe

Signed-off-by: Guodong Xu <guodong.xu@linaro.org>
---
 drivers/bluetooth/hci_ll.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/bluetooth/hci_ll.c b/drivers/bluetooth/hci_ll.c
index 3e767f245ed5..03f3b6caf242 100644
--- a/drivers/bluetooth/hci_ll.c
+++ b/drivers/bluetooth/hci_ll.c
@@ -680,6 +680,7 @@ static int hci_ti_probe(struct serdev_device *serdev)
 	struct ll_device *lldev;
 	struct nvmem_cell *bdaddr_cell;
 	u32 max_speed = 3000000;
+	int err;
 
 	lldev = devm_kzalloc(&serdev->dev, sizeof(struct ll_device), GFP_KERNEL);
 	if (!lldev)
@@ -693,6 +694,12 @@ static int hci_ti_probe(struct serdev_device *serdev)
 	if (IS_ERR(lldev->enable_gpio))
 		return PTR_ERR(lldev->enable_gpio);
 
+	err = gpiod_direction_output(lldev->enable_gpio, 0);
+	if (unlikely(err)) {
+		bt_dev_err(hu->hdev, "unable to configure enable-gpio");
+		return err;
+	}
+
 	lldev->ext_clk = devm_clk_get(&serdev->dev, "ext_clock");
 	if (IS_ERR(lldev->ext_clk) && PTR_ERR(lldev->ext_clk) != -ENOENT)
 		return PTR_ERR(lldev->ext_clk);
-- 
2.15.1

