From d425938f28d05f6f187428c7f3e02c8b97ca09c6 Mon Sep 17 00:00:00 2001
From: Roman Shaposhnik <rvs@zededa.com>
Date: Sun, 6 Jan 2019 04:30:06 +0000
Subject: [PATCH] Fixing HiKey Kconfig dependency

---
 drivers/misc/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index 8e04fc87b..6b5987cdb 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -516,6 +516,7 @@ config MISC_RTSX
 config HISI_HIKEY_USB
 	tristate "USB functionality of HiSilicon Hikey Platform"
 	depends on GPIOLIB
+        select USB_ROLE_SWITCH
 	default n
 	help
 	  If you say yes here you get support for usb functionality of HiSilicon Hikey Platform.
-- 
2.17.1

