diff --git a/drivers/pinctrl/intel/pinctrl-baytrail.c b/drivers/pinctrl/intel/pinctrl-baytrail.c
index f38d596..1e15abc 100644
--- a/drivers/pinctrl/intel/pinctrl-baytrail.c
+++ b/drivers/pinctrl/intel/pinctrl-baytrail.c
@@ -1723,6 +1723,7 @@ static int byt_gpio_probe(struct byt_gpio *vg)
 		return ret;
 	}
 
+#if 0
 	/* set up interrupts  */
 	irq_rc = platform_get_resource(vg->pdev, IORESOURCE_IRQ, 0);
 	if (irq_rc && irq_rc->start) {
@@ -1738,6 +1739,7 @@ static int byt_gpio_probe(struct byt_gpio *vg)
 					     (unsigned)irq_rc->start,
 					     byt_gpio_irq_handler);
 	}
+#endif	
 
 	return ret;
 }
