From 1e976a936c0b2d90f157bb309b8cdb167b6ad1a8 Mon Sep 17 00:00:00 2001
From: Aleksandrov Dmitriy <goodmobiledevices@gmail.com>
Date: Fri, 22 Oct 2021 03:33:44 +0300
Subject: [PATCH] board: raspberry: enable simple-framebuffer in dt

Signed-off-by: Aleksandrov Dmitriy <goodmobiledevices@gmail.com>
---
 board/raspberrypi/rpi/rpi.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index 372b26b6f2..99c6573269 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -500,8 +500,10 @@ int ft_board_setup(void *blob, struct bd_info *bd)
 	int node;
 
 	node = fdt_node_offset_by_compatible(blob, -1, "simple-framebuffer");
-	if (node < 0)
+	if (node < 0) {
 		lcd_dt_simplefb_add_node(blob);
+		lcd_dt_simplefb_enable_existing_node(blob);
+	}
 
 #ifdef CONFIG_EFI_LOADER
 	/* Reserve the spin table */
-- 
2.25.1

