From 791ad47c11dbfc11ad13137d200a33e2feaf98f1 Mon Sep 17 00:00:00 2001
From: Stefano Stabellini <sstabellini@kernel.org>
Date: Thu, 17 Dec 2020 18:27:32 -0800
Subject: [PATCH 02/21] Xen SMBIOS property add

---
 arch/arm/xen/enlighten.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/xen/enlighten.c b/arch/arm/xen/enlighten.c
index 8ad576ecd0f1..2a7ec6552f22 100644
--- a/arch/arm/xen/enlighten.c
+++ b/arch/arm/xen/enlighten.c
@@ -33,6 +33,7 @@
 #include <linux/timekeeping.h>
 #include <linux/timekeeper_internal.h>
 #include <linux/acpi.h>
+#include <linux/efi.h>
 
 #include <linux/mm.h>
 
@@ -62,6 +63,8 @@ static __read_mostly unsigned int xen_events_irq;
 uint32_t xen_start_flags;
 EXPORT_SYMBOL(xen_start_flags);
 
+uint64_t smbios_addr;
+
 int xen_unmap_domain_gfn_range(struct vm_area_struct *vma,
 			       int nr, struct page **pages)
 {
@@ -303,6 +306,11 @@ static void __init xen_dt_guest_init(void)
 	}
 
 	xen_events_irq = irq_of_parse_and_map(xen_node, 0);
+
+	if (!of_property_read_u64(xen_node, "xen,smbios", &smbios_addr)) {
+		efi.smbios = smbios_addr;
+		set_bit(EFI_CONFIG_TABLES, &efi.flags);
+	}
 }
 
 static int __init xen_guest_init(void)
-- 
2.25.1

