--- BaseTools/Source/C/Include/AArch64/ProcessorBind.h.orig
+++ BaseTools/Source/C/Include/AArch64/ProcessorBind.h
@@ -61,7 +61,9 @@
   typedef char                CHAR8;
   typedef signed char         INT8;
 
+#ifndef UINT8_MAX
   #define UINT8_MAX 0xff
+#endif
 #endif
 
 ///
--- OvmfPkg/PlatformPei/Xen.c.orig
+++ OvmfPkg/PlatformPei/Xen.c
@@ -137,6 +137,10 @@
       mXen = TRUE;
       return TRUE;
     }
+    if (!AsciiStrCmp ((CHAR8 *) Signature, "ZenZenZenZen")) {
+      mXen = TRUE;
+      return TRUE;
+    }
   }
 
   mXenLeaf = 0;
