From d586c054af76bdf7fad3039a884a496ea403ca41 Mon Sep 17 00:00:00 2001
From: Petr Fedchenkov <giggsoff@gmail.com>
Date: Thu, 15 Jul 2021 22:24:29 +0300
Subject: [PATCH] fix igd build

---
 OvmfPkg/IgdAssignmentDxe/IgdAssignment.c | 42 ------------------------
 1 file changed, 42 deletions(-)

diff --git a/OvmfPkg/IgdAssignmentDxe/IgdAssignment.c b/OvmfPkg/IgdAssignmentDxe/IgdAssignment.c
index 353facf..1142ff4 100644
--- a/OvmfPkg/IgdAssignmentDxe/IgdAssignment.c
+++ b/OvmfPkg/IgdAssignmentDxe/IgdAssignment.c
@@ -114,36 +114,6 @@ InitPciInfo (
 }
 
 
-/**
-  Format and get the debug name of a CANDIDATE_PCI_INFO structure.
-
-  param[in,out] PciInfo  If the PciInfo->Name member is an empty string, format
-                         the PCI bus address of PciInfo into PciInfo->Name.
-                         Otherwise, don't modify PciInfo.
-
-  @return                PciInfo->Name
-**/
-STATIC
-CONST CHAR8 *
-GetPciName (
-  IN OUT CANDIDATE_PCI_INFO *PciInfo
-  )
-{
-  if (PciInfo->Name[0] == '\0') {
-    AsciiSPrint (
-      PciInfo->Name,
-      sizeof PciInfo->Name,
-      "%04x:%02x:%02x.%x",
-      (UINT16)PciInfo->Segment,
-      (UINT8)PciInfo->Bus,
-      (UINT8)PciInfo->Device,
-      (UINT32)PciInfo->Function & 0xf
-      );
-  }
-  return PciInfo->Name;
-}
-
-
 /**
   Allocate memory in the 32-bit address space, with the requested UEFI memory
   type and the requested alignment.
@@ -292,8 +262,6 @@ SetupOpRegion (
              &Address
              );
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "%a: %a: failed to allocate OpRegion: %r\n",
-      __FUNCTION__, GetPciName (PciInfo), Status));
     return Status;
   }
 
@@ -316,13 +284,9 @@ SetupOpRegion (
                         &Address
                         );
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "%a: %a: failed to write OpRegion address: %r\n",
-      __FUNCTION__, GetPciName (PciInfo), Status));
     goto FreeOpRegion;
   }
 
-  DEBUG ((DEBUG_INFO, "%a: %a: OpRegion @ 0x%Lx size 0x%Lx\n", __FUNCTION__,
-    GetPciName (PciInfo), Address, (UINT64)mOpRegionSize));
   return EFI_SUCCESS;
 
 FreeOpRegion:
@@ -370,8 +334,6 @@ SetupStolenMemory (
              &Address
              );
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "%a: %a: failed to allocate stolen memory: %r\n",
-      __FUNCTION__, GetPciName (PciInfo), Status));
     return Status;
   }
 
@@ -391,13 +353,9 @@ SetupStolenMemory (
                         &Address
                         );
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "%a: %a: failed to write stolen memory address: %r\n",
-      __FUNCTION__, GetPciName (PciInfo), Status));
     goto FreeStolenMemory;
   }
 
-  DEBUG ((DEBUG_INFO, "%a: %a: stolen memory @ 0x%Lx size 0x%Lx\n",
-    __FUNCTION__, GetPciName (PciInfo), Address, (UINT64)mBdsmSize));
   return EFI_SUCCESS;
 
 FreeStolenMemory:
-- 
2.30.2

