diff --git a/pkg/pillar/vendor/github.com/shirou/gopsutil/process/process_linux.go b/pkg/pillar/vendor/github.com/shirou/gopsutil/process/process_linux.go
index 8ae99ff29..709bf8172 100644
--- a/pkg/pillar/vendor/github.com/shirou/gopsutil/process/process_linux.go
+++ /vendor/github.com/shirou/gopsutil/process/process_linux.go
@@ -1051,8 +1051,12 @@ func (p *Process) fillFromStatusWithContext(ctx context.Context) error {
 					}
 				}
 			}
+			// Ensure we have a copy and not reference into slice
+			p.name = string([]byte(p.name))
 		case "State":
 			p.status = value[0:1]
+			// Ensure we have a copy and not reference into slice
+			p.status = string([]byte(p.status))
 		case "PPid", "Ppid":
 			pval, err := strconv.ParseInt(value, 10, 32)
 			if err != nil {
