diff --git a/pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go b/pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go
index 4dd052944..4a936cc8e 100644
--- a/pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go
+++ /vendor/github.com/digitalocean/go-qemu/qmp/socket.go
@@ -185,7 +185,15 @@ func (mon *SocketMonitor) listen(r io.Reader, events chan<- Event, stream chan<-
 	}
 
 	if err := scanner.Err(); err != nil {
-		stream <- streamResponse{err: err}
+		// In case stream reader went away we wait for a bit
+		waitTimer := time.NewTimer(3 * time.Second)
+		defer waitTimer.Stop()
+		select {
+		case <-waitTimer.C:
+			// Do nothing
+		case stream <- streamResponse{err: err}:
+			// Done
+		}
 	}
 }
 
