From 162251266437b5e2282e48557934065a554b34d2 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Thu, 15 Oct 2020 20:41:31 -0400
Subject: [PATCH] Fix build warning

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 tools/libxl/libxlu_pci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/libxl/libxlu_pci.c b/tools/libxl/libxlu_pci.c
index 7947687..595c5ea 100644
--- a/tools/libxl/libxlu_pci.c
+++ b/tools/libxl/libxlu_pci.c
@@ -48,7 +48,7 @@ static int pcidev_struct_fill(libxl_device_pci *pcidev, unsigned int domain,
 int xlu_pci_parse_bdf(XLU_Config *cfg, libxl_device_pci *pcidev, const char *str)
 {
     unsigned state = STATE_DOMAIN;
-    unsigned dom, bus, dev, func, vslot = 0;
+    unsigned dom = 0, bus = 0, dev = 0, func = 0, vslot = 0;
     char *buf2, *tok, *ptr, *end, *optkey = NULL;
 
     if ( NULL == (buf2 = ptr = strdup(str)) )
-- 
2.26.2

