
diff --git a/tools/firmware/hvmloader/hvmloader.c b/tools/firmware/hvmloader/hvmloader.c
index 6d23150fc9..038d76e3ab 100644
--- a/tools/firmware/hvmloader/hvmloader.c
+++ b/tools/firmware/hvmloader/hvmloader.c
@@ -136,10 +136,10 @@ static void init_hypercalls(void)
 
         if ( !strcmp("XenVMMXenVMM", signature) )
             break;
+       if ( !strcmp("ZenZenZenZen", signature) )
+               break;
     }
 
-    BUG_ON(strcmp("XenVMMXenVMM", signature) || ((eax - base) < 2));
-
     /* Fill in hypercall transfer pages. */
     cpuid(base + 2, &eax, &ebx, &ecx, &edx);
     for ( i = 0; i < eax; i++ )
diff --git a/tools/libs/light/libxl_create.c b/tools/libs/light/libxl_create.c
index bfc9149096..6a0fedd29c 100644
--- a/tools/libs/light/libxl_create.c
+++ b/tools/libs/light/libxl_create.c
@@ -308,6 +308,8 @@ int libxl__domain_build_info_setdefault(libxl__gc *gc,
         libxl_defbool_setdefault(&b_info->u.hvm.acpi_laptop_slate,  false);
         libxl_defbool_setdefault(&b_info->u.hvm.nx,                 true);
         libxl_defbool_setdefault(&b_info->u.hvm.viridian,           false);
+        libxl_defbool_setdefault(&b_info->u.hvm.spoof_viridian,     false);
+        libxl_defbool_setdefault(&b_info->u.hvm.spoof_xen,          false);
         libxl_defbool_setdefault(&b_info->u.hvm.hpet,               true);
         libxl_defbool_setdefault(&b_info->u.hvm.vpt_align,          true);
         libxl_defbool_setdefault(&b_info->u.hvm.usb,                false);
@@ -1791,6 +1793,10 @@ static void domcreate_launch_dm(libxl__egc *egc, libxl__multidev *multidev,
         console.backend_domid = state->console_domid;
         libxl__device_console_add(gc, domid, &console, state, &device);
         libxl__device_console_dispose(&console);
+        LOG(DEBUG, "Checking spoofing for guest (domid %d): xen %d, vir %d", domid,
+                                        libxl_defbool_val(d_config->b_info.u.hvm.spoof_xen),
+                                        libxl_defbool_val(d_config->b_info.u.hvm.spoof_viridian)
+                         );
 
         dcs->sdss.dm.guest_domid = domid;
         if (libxl_defbool_val(d_config->b_info.device_model_stubdomain))
diff --git a/tools/libs/light/libxl_types.idl b/tools/libs/light/libxl_types.idl
index d64a573ff3..d5e7f7b8a8 100644
--- a/tools/libs/light/libxl_types.idl
+++ b/tools/libs/light/libxl_types.idl
@@ -664,6 +664,8 @@ libxl_domain_build_info = Struct("domain_build_info",[
                                        ("viridian",         libxl_defbool),
                                        ("viridian_enable",  libxl_bitmap),
                                        ("viridian_disable", libxl_bitmap),
+                                       ("spoof_viridian",   libxl_defbool),
+                                       ("spoof_xen",        libxl_defbool),
                                        ("timeoffset",       string),
                                        ("hpet",             libxl_defbool),
                                        ("vpt_align",        libxl_defbool),
diff --git a/tools/libs/light/libxl_x86.c b/tools/libs/light/libxl_x86.c
index 60d4e8661c..a079b3a316 100644
--- a/tools/libs/light/libxl_x86.c
+++ b/tools/libs/light/libxl_x86.c
@@ -425,6 +425,16 @@ static int hvm_set_conf_params(libxl__gc *gc, uint32_t domid,
             LOG(ERROR, "Couldn't set HVM_PARAM_VPT_ALIGN");
             goto out;
         }
+        if (xc_hvm_param_set(xch, domid, HVM_PARAM_SPOOF_XEN,
+                             libxl_defbool_val(info->u.hvm.spoof_xen))) {
+            LOG(ERROR, "Couldn't set HVM_PARAM_SPOOF_XEN");
+            goto out;
+        }
+        if (xc_hvm_param_set(xch, domid, HVM_PARAM_SPOOF_VIRIDIAN,
+                             libxl_defbool_val(info->u.hvm.spoof_viridian))) {
+            LOG(ERROR, "Couldn't set HVM_PARAM_SPOOF_VIRIDIAN");
+            goto out;
+        }
         if (info->u.hvm.mca_caps &&
             xc_hvm_param_set(CTX->xch, domid, HVM_PARAM_MCA_CAP,
                              info->u.hvm.mca_caps)) {
diff --git a/tools/misc/xen-detect.c b/tools/misc/xen-detect.c
index 4f74c41557..26d02b291b 100644
--- a/tools/misc/xen-detect.c
+++ b/tools/misc/xen-detect.c
@@ -85,6 +85,8 @@ static int check_for_xen(int pv_context)
 
         if ( !strcmp("XenVMMXenVMM", signature) && (regs[0] >= (base + 2)) )
             goto found;
+        if ( !strcmp("ZenZenZenZen", signature) && (regs[0] >= (base + 2)) )
+            goto found;
 
         /* Higher base addresses are possible only with HVM. */
         if ( pv_context )
diff --git a/tools/xl/xl_parse.c b/tools/xl/xl_parse.c
index af86d3186d..0643bf0775 100644
--- a/tools/xl/xl_parse.c
+++ b/tools/xl/xl_parse.c
@@ -1873,6 +1873,10 @@ void parse_config_data(const char *config_source,
         xlu_cfg_get_defbool(config, "apic", &b_info->apic, 0);
         xlu_cfg_get_defbool(config, "hvm_pirq", &b_info->u.hvm.pirq, 0);
 
+        /* For spoofing Xen */
+        xlu_cfg_get_defbool(config, "spoof_xen", &b_info->u.hvm.spoof_xen, 0);
+        xlu_cfg_get_defbool(config, "spoof_viridian", &b_info->u.hvm.spoof_viridian, 0);
+
         switch (xlu_cfg_get_list(config, "viridian",
                                  &viridian, &num_viridian, 1))
         {
diff --git a/xen/arch/x86/cpuid.c b/xen/arch/x86/cpuid.c
index b63a82dd38..995b960891 100644
--- a/xen/arch/x86/cpuid.c
+++ b/xen/arch/x86/cpuid.c
@@ -58,9 +58,17 @@ static void cpuid_hypervisor_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = base + limit; /* Largest leaf */
-        res->b = XEN_CPUID_SIGNATURE_EBX;
-        res->c = XEN_CPUID_SIGNATURE_ECX;
-        res->d = XEN_CPUID_SIGNATURE_EDX;
+        if (!d->arch.hvm.spoof_xen) {
+            printk("cpuid_hypervisor_leaves - real id. domid %d\n", d->domain_id);
+            res->b = XEN_CPUID_SIGNATURE_EBX;
+            res->c = XEN_CPUID_SIGNATURE_ECX;
+            res->d = XEN_CPUID_SIGNATURE_EDX;
+        } else {
+            printk("cpuid_hypervisor_leaves - spoofed id. domid %d\n", d->domain_id);
+            res->b = ZEN_CPUID_SIGNATURE_EBX;
+            res->c = ZEN_CPUID_SIGNATURE_ECX;
+            res->d = ZEN_CPUID_SIGNATURE_EDX;
+        }
         break;
 
     case 1:
diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index 0c60faa39d..13349eebd9 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -4358,6 +4358,14 @@ static int hvm_set_param(struct domain *d, uint32_t index, uint64_t value)
     case HVM_PARAM_IOREQ_SERVER_PFN:
         d->arch.hvm.ioreq_gfn.base = value;
         break;
+    case HVM_PARAM_SPOOF_XEN:
+        printk("spoof_xen %" PRId64 " - curr dom %d, dom %d\n", value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_xen = value;
+        break;
+    case HVM_PARAM_SPOOF_VIRIDIAN:
+        printk("spoof_vir %" PRId64 " - curr dom %d, dom %d\n", value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_viridian = value;
+        break;
     case HVM_PARAM_NR_IOREQ_SERVER_PAGES:
     {
         unsigned int i;
diff --git a/xen/arch/x86/hvm/viridian/viridian.c b/xen/arch/x86/hvm/viridian/viridian.c
index f79cffcb37..d94bb9d9a3 100644
--- a/xen/arch/x86/hvm/viridian/viridian.c
+++ b/xen/arch/x86/hvm/viridian/viridian.c
@@ -124,9 +124,13 @@ void cpuid_viridian_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = 0x40000006; /* Maximum leaf */
-        memcpy(&res->b, "Micr", 4);
-        memcpy(&res->c, "osof", 4);
-        memcpy(&res->d, "t Hv", 4);
+	printk("wetware cpuid_viridian_leaves spoof_vir\n"); // "Wetware Labs"
+        //memcpy(&res->b, "Micr", 4);
+        //memcpy(&res->c, "osof", 4);
+        //memcpy(&res->d, "t Hv", 4);
+        memcpy(&res->b, "Wetw", 4);
+        memcpy(&res->c, "era ", 4);
+        memcpy(&res->d, "Labs", 4);
         break;
 
     case 1:
diff --git a/xen/arch/x86/include/asm/hvm/domain.h b/xen/arch/x86/include/asm/hvm/domain.h
index 333501d5f2..c6c3714d05 100644
--- a/xen/arch/x86/include/asm/hvm/domain.h
+++ b/xen/arch/x86/include/asm/hvm/domain.h
@@ -105,6 +105,9 @@ struct hvm_domain {
 
     bool                   is_s3_suspended;
 
+    bool                  spoof_xen;
+    bool                  spoof_viridian;
+
     /* Compatibility setting for a bug in x2APIC LDR */
     bool bug_x2apic_ldr_vcpu_id;
 
diff --git a/xen/include/public/arch-x86/cpuid.h b/xen/include/public/arch-x86/cpuid.h
index 3bb0dd249f..bc71df630a 100644
--- a/xen/include/public/arch-x86/cpuid.h
+++ b/xen/include/public/arch-x86/cpuid.h
@@ -36,6 +36,10 @@
 #define XEN_CPUID_SIGNATURE_ECX 0x65584d4d /* "MMXe" */
 #define XEN_CPUID_SIGNATURE_EDX 0x4d4d566e /* "nVMM" */
 
+#define ZEN_CPUID_SIGNATURE_EBX 0x5A6e655A /* "ZenZ" */
+#define ZEN_CPUID_SIGNATURE_ECX 0x655A6e65 /* "enZe" */
+#define ZEN_CPUID_SIGNATURE_EDX 0x6e655A6e /* "nZen" */
+
 /*
  * Leaf 2 (0x40000x01)
  * EAX[31:16]: Xen major version.
diff --git a/xen/include/public/hvm/params.h b/xen/include/public/hvm/params.h
index 99c40b4287..5bec66173a 100644
--- a/xen/include/public/hvm/params.h
+++ b/xen/include/public/hvm/params.h
@@ -290,7 +290,9 @@
 #define HVM_PARAM_MCA_CAP 38
 #define XEN_HVM_MCA_CAP_LMCE   (xen_mk_ullong(1) << 0)
 #define XEN_HVM_MCA_CAP_MASK   XEN_HVM_MCA_CAP_LMCE
+#define HVM_PARAM_SPOOF_XEN 39
 
-#define HVM_NR_PARAMS 39
+#define HVM_PARAM_SPOOF_VIRIDIAN 40
+#define HVM_NR_PARAMS 41
 
 #endif /* __XEN_PUBLIC_HVM_PARAMS_H__ */
