diff --git a/pam/pam.h b/pam/pam.h
index 09afb2e..d1e190b 100644
--- a/pam/pam.h
+++ b/pam/pam.h
@@ -23,7 +23,7 @@
 #include <security/pam_appl.h>
 
 // Conversation that will call back into Go code when appropriate.
-const struct pam_conv *goConv;
+extern const struct pam_conv *goConv;
 
 // CleaupFuncs are used to cleanup specific PAM data.
 typedef void (*CleanupFunc)(pam_handle_t *pamh, void *data, int error_status);
