From 5836841a7e3dc3045c927835358e3ad819f123ef Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Mon, 21 Dec 2020 06:16:05 +0300
Subject: [PATCH] rte_config: Do not call backtrace()

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 config/rte_config.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/rte_config.h b/config/rte_config.h
index a0b5160ff..2d8141944 100644
--- a/config/rte_config.h
+++ b/config/rte_config.h
@@ -38,7 +38,6 @@
 #define RTE_MAX_MEMZONE 2560
 #define RTE_MAX_TAILQ 32
 #define RTE_LOG_DP_LEVEL RTE_LOG_INFO
-#define RTE_BACKTRACE 1
 #define RTE_MAX_VFIO_CONTAINERS 64
 
 /* bsd module defines */
-- 
2.25.1

