diff --git a/Makefile.am b/Makefile.am
index 5e53f62..717eb3a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,7 +14,7 @@ include src_vars.mk
 ACLOCAL_AMFLAGS = -I m4 --install
 
 INCLUDE_DIRS = -I$(top_srcdir)/tools -I$(top_srcdir)/lib
-LIB_COMMON := lib/libcommon.a
+LIB_COMMON := lib/libcommon.la
 
 AM_CFLAGS := \
     $(INCLUDE_DIRS) $(EXTRA_CFLAGS) $(TSS2_ESYS_CFLAGS) $(TSS2_MU_CFLAGS) \
@@ -40,9 +40,9 @@ bin_PROGRAMS += tools/fapi/tss2
 
 endif
 
-noinst_LIBRARIES = $(LIB_COMMON)
-lib_libcommon_a_SOURCES = $(LIB_SRC)
-lib_libcommon_a_CFLAGS = -fPIC $(AM_CFLAGS)
+lib_LTLIBRARIES = $(LIB_COMMON)
+lib_libcommon_la_SOURCES = $(LIB_SRC)
+lib_libcommon_la_CFLAGS = -fPIC -shared $(AM_CFLAGS)
 
 tools_fapi_tss2_CFLAGS = $(FAPI_CFLAGS) -DTSS2_TOOLS_MAX="$(words $(tss2_tools))"
 tools_fapi_tss2_LDFLAGS = $(EXTRA_LDFLAGS) $(TSS2_FAPI_LIBS)
