diff --git a/actions/callback.go b/actions/callback.go
index f15893d..65ced07 100644
--- a/actions/callback.go
+++ b/actions/callback.go
@@ -102,8 +102,9 @@ func unwrapProtectorKey(info ProtectorInfo, keyFn KeyFunc) (*crypto.Key, error)
 		case crypto.ErrBadAuth:
 			// After the first failure, we let the callback know we are retrying.
 			log.Printf("invalid wrapping key for protector %s", info.Descriptor())
-			retry = true
-			continue
+			return nil, err
+			//retry = true
+			//continue
 		default:
 			return nil, err
 		}
diff --git a/cmd/fscrypt/keys.go b/cmd/fscrypt/keys.go
index 872ca2a..e3304bc 100644
--- a/cmd/fscrypt/keys.go
+++ b/cmd/fscrypt/keys.go
@@ -33,7 +33,6 @@ import (
 	"github.com/google/fscrypt/actions"
 	"github.com/google/fscrypt/crypto"
 	"github.com/google/fscrypt/metadata"
-	"github.com/google/fscrypt/pam"
 )
 
 // The file descriptor for standard input
@@ -124,29 +123,8 @@ func makeKeyFunc(supportRetry, shouldConfirm bool, prefix string) actions.KeyFun
 
 		switch info.Source() {
 		case metadata.SourceType_pam_passphrase:
-			prompt := fmt.Sprintf("Enter %slogin passphrase for %s: ",
-				prefix, formatUsername(info.UID()))
-			key, err := getPassphraseKey(prompt)
-			if err != nil {
-				return nil, err
-			}
-
-			// To confirm, check that the passphrase is the user's
-			// login passphrase.
-			if shouldConfirm {
-				username, err := usernameFromID(info.UID())
-				if err != nil {
-					key.Wipe()
-					return nil, err
-				}
-
-				err = pam.IsUserLoginToken(username, key, quietFlag.Value)
-				if err != nil {
-					key.Wipe()
-					return nil, err
-				}
-			}
-			return key, nil
+			fmt.Println("This source type is not supported on this system")
+			return nil, ErrInvalidSource
 
 		case metadata.SourceType_custom_passphrase:
 			prompt := fmt.Sprintf("Enter %scustom passphrase for protector %q: ",
