--- xen/tools/hotplug/Linux/locking.sh	2019-07-19 12:54:37.000000000 -0700
+++ xen/tools/hotplug/Linux/locking.sh	2020-03-21 00:14:37.000000000 -0700
@@ -56,18 +56,8 @@ claim_lock()
         # WW.XXX
         # YY.ZZZ
         # which need to be separated and compared.
-        if stat=$( stat -L -c '%D.%i' - $_lockfile 0<&$_lockfd 2>/dev/null )
-        then
-            local file_stat
-            local fd_stat
-
-            # match on literal newline
-            fd_stat=${stat%
-*}
-            file_stat=${stat#*
-}
-            if [ "$fd_stat" = "$file_stat" ] ; then break; fi
-        fi
+        rightfile=$(rightfile "$_lockfile" <&$_lockfd)
+        if [ x$rightfile = xy ]; then break; fi
         # Some versions of bash appear to be buggy if the same
         # $_lockfile is opened repeatedly. Close the current fd here.
         eval "exec $_lockfd<&-"
