From cfe59eced0d6f3485b2187e3a14101802f2a4ac2 Mon Sep 17 00:00:00 2001
From: Aleksandrov Dmitriy <goodmobiledevices@gmail.com>
Date: Sun, 30 Oct 2022 18:20:46 -0100
Subject: [PATCH] board: raspberry: enable simple-framebuffer in dt

Signed-off-by: Aleksandrov Dmitriy <goodmobiledevices@gmail.com>
---
 board/raspberrypi/rpi/rpi.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index 17b8108cc8..caebf2190f 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -509,8 +509,10 @@ int ft_board_setup(void *blob, struct bd_info *bd)
 	int node;
 
 	node = fdt_node_offset_by_compatible(blob, -1, "simple-framebuffer");
-	if (node < 0)
+	if (node < 0) {
 		fdt_simplefb_add_node(blob);
+		fdt_simplefb_enable_existing_node(blob);
+	}
 
 #ifdef CONFIG_EFI_LOADER
 	/* Reserve the spin table */
-- 
2.25.1

