--- images/rootfs.yml.in	2022-11-08 16:23:13.977367851 +0000
+++ images/rootfs-acrn.yml.in	2022-11-08 16:23:14.065366837 +0000
@@ -1,5 +1,5 @@
 kernel:
-  image: KERNEL_TAG
+  image: ACRN_KERNEL_TAG
   cmdline: "rootdelay=3"
 init:
   - linuxkit/init:8f1e6a0747acbbb4d7e24dc98f97faa8d1c6cec7
@@ -10,7 +10,7 @@
   - DOM0ZTOOLS_TAG
   - GRUB_TAG
   - FW_TAG
-  - XEN_TAG
+  - ACRN_TAG
   - GPTTOOLS_TAG
   - KEXEC_TAG
 onboot:
