--- images/rootfs.yml.in
+++ images/rootfs-adam-kvm.yml.in
@@ -30,6 +30,12 @@
      image: NEWLOGD_TAG
      cgroupsPath: /eve/services/newlogd
      oomScoreAdj: -999
+   - name: adam
+     image: lfedge/adam:latest
+     binds:
+        - /var/persist:/persist
+        - /var/config:/config
+     command: ["/bin/eve-embedded.sh"]
    - name: debug
      image: DEBUG_TAG
      cgroupsPath: /eve/services/debug
