--- a/src/ifdown.c	2018-03-19 10:36:45.191867674 +0000
+++ b/src/ifdown.c	2018-03-19 10:36:58.158908584 +0000
@@ -11,6 +11,7 @@
 #include "config.h"
 #endif
 
+#include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -20,7 +21,6 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/time.h>
-#include <sys/errno.h>
 
 #include <net/if.h>
 #include <netinet/in.h>
