diff -uprN a/main/musl/APKBUILD b/main/musl/APKBUILD
--- a/main/musl/APKBUILD	2024-01-26 21:40:51.000000000 +0100
+++ b/main/musl/APKBUILD	2025-07-24 13:45:10.413536643 +0200
@@ -19,8 +19,8 @@ nocc)	pkgname="musl-dev"; subpackages=""
 nolibc) ;;
 *)	subpackages="$subpackages $pkgname-utils";;
 esac
-_commit="v1.2.3"
-source="musl-$_commit.tar.gz::https://git.musl-libc.org/cgit/musl/snapshot/$_commit.tar.gz
+_commit="1.2.3"
+source="musl-$_commit.tar.gz::http://deb.debian.org/debian/pool/main/m/musl/musl_$_commit.orig.tar.gz
 	handle-aux-at_base.patch

 	0001-fix-mishandling-of-errno-in-getaddrinfo-AI_ADDRCONFI.patch
@@ -48,7 +48,7 @@ source="musl-$_commit.tar.gz::https://gi
 #   1.1.15-r4:
 #     - CVE-2016-8859

-builddir="$srcdir"/$_commit
+builddir="$srcdir"/musl-$_commit

 build() {
 	[ "$BOOTSTRAP" = "nocc" ] && return 0
@@ -174,7 +174,7 @@ compat() {
 }

 sha512sums="
-9a1b8f9208d99582ac00e3c46c829aa1fad3b7f09aa7d6822f02f25542453d3d977c69519ad01430b8dd624ac9dc70f47d611e36aefd7fef31ea47a7679e3111  musl-v1.2.3.tar.gz
+9332f713d3eb7de4369bc0327d99252275ee52abf523ee34b894b24a387f67579787f7c72a46cf652e090cffdb0bc3719a4e7b84dca66890b6a37f12e8ad089c  musl-1.2.3.tar.gz
 a76f79b801497ad994746cf82bb6eaf86f9e1ae646e6819fbae8532a7f4eee53a96ac1d4e789ec8f66aea2a68027b0597f7a579b3369e01258da8accfce41370  handle-aux-at_base.patch
 63a7b65c2f508b0811f172d3b55da27a4b372653e01339d2f9e2b81bd61e7848ab317a6676a9440f50695658043cdede24881f8545f96bb291dd1f9b669ac08d  0001-fix-mishandling-of-errno-in-getaddrinfo-AI_ADDRCONFI.patch
 b937a2c7f5adc51d23c731436b97152afbb147f98a4f33fa644fe2fe1aebea7da4519da4eb96acd73426660ce5dfe70cc27b8868e95a0322d74bc7aff4d8bddb  0001-fix-return-value-of-gethostby-name-2-addr-with-no-re.patch
