From 54fe3230d62150b5b74a379786c6627d50ab6109 Mon Sep 17 00:00:00 2001
From: Petr Fedchenkov <giggsoff@gmail.com>
Date: Thu, 15 Jul 2021 22:15:03 +0300
Subject: [PATCH] patch xen PlatformPei

---
 OvmfPkg/XenPlatformPei/Xen.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/OvmfPkg/XenPlatformPei/Xen.c b/OvmfPkg/XenPlatformPei/Xen.c
index 7f00eef11e..8f31a79d98 100644
--- a/OvmfPkg/XenPlatformPei/Xen.c
+++ b/OvmfPkg/XenPlatformPei/Xen.c
@@ -261,6 +261,9 @@ XenDetect (
     if (!AsciiStrCmp ((CHAR8 *)Signature, "XenVMMXenVMM")) {
       return TRUE;
     }
+    if (!AsciiStrCmp ((CHAR8 *) Signature, "ZenZenZenZen")) {
+      return TRUE;
+    }
   }
 
   mXenLeaf = 0;
-- 
2.25.1

