From 62c3e9c83b72dc202d6b8130a867adf4f0e715bb Mon Sep 17 00:00:00 2001
From: Roman Shaposhnik <rvs@zededa.com>
Date: Fri, 2 Aug 2019 11:00:18 -0700
Subject: [PATCH 1/2] Adding STAGE1_XL_OPTS and fixing one nit

---
 files/run  | 7 ++++---
 files/stop | 4 +---
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/files/run b/files/run
index ab63fa0..529270b 100755
--- a/files/run
+++ b/files/run
@@ -58,7 +58,7 @@ echo "ramdisk='$workpath/initrd'" >> $outconfig
 echo "p9=[ 'tag=share_dir,security_model=none,path=$mountpoint' ]" >> $outconfig
 
 if [ -f "$STAGE1_SEED_XL_CFG" ] ; then
-    grep -Ev '^(disk|type)' < "$STAGE1_SEED_XL_CFG" >> $outconfig
+    grep -Ev '^(disk|bootloader|root|extra|kernel)' < "$STAGE1_SEED_XL_CFG" >> $outconfig
     echo extra=\'console=hvc0 root=9p\' >> $outconfig
 else	
     echo "memory = 1024" >> $outconfig
@@ -83,6 +83,7 @@ else
 fi    
 
 echo $cmdline > $mountpoint/cmdline
-xl create $gargs $workpath/out/test.0
-domid=`xl list | grep "$uuid" | awk '{print$2}'`
+xl create $gargs $outconfig $STAGE1_XL_OPTS
+domname=$(sed -ne '/^name *=/s/^.*"\(.*\)"/\1/p' < "$outconfig")
+domid=`xl list | grep "$domname" | awk '{print$2}'`
 echo $domid > $pod/pid
diff --git a/files/stop b/files/stop
index e3e69e3..5ca1cbe 100755
--- a/files/stop
+++ b/files/stop
@@ -1,6 +1,4 @@
 #!/bin/bash
 
-domain=$(sed -ne '/^name *=/s/^.*"\(.*\)"/\1/p' < "/var/lib/rkt/pods/run/$1/stage1/rootfs/out/test.0")
-
-xl destroy $domain &>/dev/null
+xl destroy $(cat "/var/lib/rkt/pods/run/$1/pid") &>/dev/null
 exit 0
-- 
2.15.1

