--- a/kernel/init-initrd
+++ b/kernel/init-initrd
@@ -34,6 +34,8 @@
 mount -o bind /dev /mnt/rootfs/dev
 mount -o bind /sys /mnt/rootfs/sys
 mount -o bind /proc /mnt/rootfs/proc
+mount -t devpts -o gid=5,mode=0620,noexec,nosuid devpts /mnt/rootfs/dev/pts
+mount -t tmpfs -o nodev,nosuid,noexec shm /mnt/rootfs/dev/shm
 
 ip=`cat /proc/cmdline | grep -o '\bip=[^ ]*' | cut -d = -f 2`
 gw=`cat /proc/cmdline | grep -o '\bgw=[^ ]*' | cut -d = -f 2`
--- a/kernel/make-initrd
+++ b/kernel/make-initrd
@@ -12,6 +11,8 @@
 mkdir -p $tmpdir/sbin
 mkdir -p $tmpdir/etc
 mkdir -p $tmpdir/dev
+mkdir -p $tmpdir/dev/pts
+mkdir -p $tmpdir/dev/shm
 mkdir -p $tmpdir/proc
 mkdir -p $tmpdir/sys
 mkdir -p $tmpdir/lib
