diff --git a/files/run b/files/run
index 4361c51..c0c49a2 100644
--- a/files/run
+++ b/files/run
@@ -68,13 +68,18 @@
 
 if [ -f "$STAGE1_SEED_XL_CFG" ] ; then
     grep -Ev '^(bootloader|root|extra|kernel)' < "$STAGE1_SEED_XL_CFG" >> $outconfig
-    if test $dhcp -eq 0
-    then
-        echo extra=\'console=hvc0 root=9p\' >> $outconfig
+    EXTRA="root=9p"
+    if grep -q 'type.*=.*hvm' "$STAGE1_SEED_XL_CFG" ; then
+       EXTRA="$EXTRA console=tty0"
     else
+       EXTRA="$EXTRA console=hvc0"
+    fi
+    if test $dhcp -ne 0
+    then
         cp $stage1/udhcpc_script.sh $stage2
-        echo extra=\'console=hvc0 root=9p dhcp=1\' >> $outconfig
+        EXTRA="$EXTRA dhcp=1"
     fi
+    echo extra=\'$EXTRA\' >> $outconfig
 else	
     echo "memory = 1024" >> $outconfig
     echo "vcpus = 2" >> $outconfig
