diff --git a/arch/x86/time.c b/arch/x86/time.c
index 3b451525a9..cae02654b9 100644
--- a/arch/x86/time.c
+++ b/arch/x86/time.c
@@ -1597,11 +1597,7 @@ static void __init probe_wallclock(void)
         wallclock_source = WALLCLOCK_CMOS;
         return;
     }
-    if ( efi_enabled(EFI_RS) && efi_get_time() )
-    {
-        wallclock_source = WALLCLOCK_EFI;
-        return;
-    }
+    /* EVE: skip EFI wallclock — gettime via EFI considered harmful */
 
     panic("No usable wallclock found, probed:%s%s%s\n%s",
           !opt_cmos_rtc_probe && !efi_enabled(EFI_RS) ? " None" : "",
