diff --git a/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index 4723f5d..311e8a5 100644
--- a/arch/x86/hvm/hvm.c
+++ b/arch/x86/hvm/hvm.c
@@ -4274,6 +4274,14 @@ static int hvmop_set_param(
     case HVM_PARAM_IOREQ_SERVER_PFN:
         d->arch.hvm.ioreq_gfn.base = a.value;
         break;
+    case HVM_PARAM_SPOOF_XEN:
+        printk("spoof_xen %" PRId64 " - curr dom %d, dom %d\n", a.value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_xen = a.value;
+        break;
+    case HVM_PARAM_SPOOF_VIRIDIAN:
+        printk("spoof_vir %" PRId64 " - curr dom %d, dom %d\n", a.value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_viridian = a.value;
+        break;
     case HVM_PARAM_NR_IOREQ_SERVER_PAGES:
     {
         unsigned int i;
diff --git a/arch/x86/hvm/viridian/viridian.c b/xen/arch/x86/hvm/viridian/viridian.c
index 44c8e6c..b45bdbf 100644
--- a/arch/x86/hvm/viridian/viridian.c
+++ b/arch/x86/hvm/viridian/viridian.c
@@ -121,9 +121,13 @@ void cpuid_viridian_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = 0x40000006; /* Maximum leaf */
-        memcpy(&res->b, "Micr", 4);
-        memcpy(&res->c, "osof", 4);
-        memcpy(&res->d, "t Hv", 4);
+	printk("wetware cpuid_viridian_leaves spoof_vir\n"); // "Wetware Labs"
+        //memcpy(&res->b, "Micr", 4);
+        //memcpy(&res->c, "osof", 4);
+        //memcpy(&res->d, "t Hv", 4);
+        memcpy(&res->b, "Wetw", 4);
+        memcpy(&res->c, "era ", 4);
+        memcpy(&res->d, "Labs", 4);
         break;
 
     case 1:
diff --git a/arch/x86/traps.c b/xen/arch/x86/traps.c
index 97499a0..b500095 100644
--- a/arch/x86/traps.c
+++ b/arch/x86/traps.c
@@ -892,9 +892,19 @@ void cpuid_hypervisor_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = base + limit; /* Largest leaf */
-        res->b = XEN_CPUID_SIGNATURE_EBX;
-        res->c = XEN_CPUID_SIGNATURE_ECX;
-        res->d = XEN_CPUID_SIGNATURE_EDX;
+        if (!d->arch.hvm.spoof_xen ) {
+            printk("cpuid_hypervisor_leaves - real id. domid %d\n",d->domain_id);
+               res->b = XEN_CPUID_SIGNATURE_EBX;
+               res->c = XEN_CPUID_SIGNATURE_ECX;
+               res->d = XEN_CPUID_SIGNATURE_EDX;
+        } else
+        {
+            printk("cpuid_hypervisor_leaves - spoofed id. domid %d\n",d->domain_id);
+               res->b = ZEN_CPUID_SIGNATURE_EBX;
+               res->c = ZEN_CPUID_SIGNATURE_ECX;
+               res->d = ZEN_CPUID_SIGNATURE_EDX;
+        }
+
         break;
 
     case 1:
diff --git a/include/asm-x86/hvm/domain.h b/xen/include/asm-x86/hvm/domain.h
index bcc5621..a506710 100644
--- a/include/asm-x86/hvm/domain.h
+++ b/include/asm-x86/hvm/domain.h
@@ -159,6 +159,8 @@ struct hvm_domain {
     bool_t                 mem_sharing_enabled;
     bool_t                 qemu_mapcache_invalidate;
     bool_t                 is_s3_suspended;
+    bool_t                 spoof_xen;
+    bool_t                 spoof_viridian;
 
     /*
      * TSC value that VCPUs use to calculate their tsc_offset value.
diff --git a/include/public/arch-x86/cpuid.h b/xen/include/public/arch-x86/cpuid.h
index ce46305..7e63c55 100644
--- a/include/public/arch-x86/cpuid.h
+++ b/include/public/arch-x86/cpuid.h
@@ -53,6 +53,10 @@
 #define XEN_CPUID_SIGNATURE_ECX 0x65584d4d /* "MMXe" */
 #define XEN_CPUID_SIGNATURE_EDX 0x4d4d566e /* "nVMM" */
 
+#define ZEN_CPUID_SIGNATURE_EBX 0x5A6e655A /* "ZenZ" */
+#define ZEN_CPUID_SIGNATURE_ECX 0x655A6e65 /* "enZe" */
+#define ZEN_CPUID_SIGNATURE_EDX 0x6e655A6e /* "nZen" */
+
 /*
  * Leaf 2 (0x40000x01)
  * EAX[31:16]: Xen major version.
diff --git a/include/public/hvm/params.h b/xen/include/public/hvm/params.h
index 36832e4..2be4178 100644
--- a/include/public/hvm/params.h
+++ b/include/public/hvm/params.h
@@ -302,7 +302,10 @@
 #define HVM_PARAM_MCA_CAP 38
 #define XEN_HVM_MCA_CAP_LMCE   (xen_mk_ullong(1) << 0)
 #define XEN_HVM_MCA_CAP_MASK   XEN_HVM_MCA_CAP_LMCE
+#define HVM_PARAM_SPOOF_XEN    39
 
-#define HVM_NR_PARAMS 39
+#define HVM_PARAM_SPOOF_VIRIDIAN    40
+
+#define HVM_NR_PARAMS          41
 
 #endif /* __XEN_PUBLIC_HVM_PARAMS_H__ */
