--- xen/tools/hotplug/Linux/locking.sh	2019-07-19 12:54:37.000000000 -0700
+++ xen/tools/hotplug/Linux/locking.sh	2020-03-21 00:14:37.000000000 -0700
@@ -51,12 +51,7 @@
         # guaranteed that our stat(2) won't lose the race with an
         # rm(1) between reading the synthetic link and traversing the
         # file system to find the inum.  Perl is very fast so use that.
-        rightfile=$( perl -e '
-            open STDIN, "<&'$_lockfd'" or die $!;
-            my $fd_inum = (stat STDIN)[1]; die $! unless defined $fd_inum;
-            my $file_inum = (stat $ARGV[0])[1];
-            print "y\n" if $fd_inum eq $file_inum;
-                             ' "$_lockfile" )
+        rightfile=$(rightfile "$_lockfile" <&$_lockfd)
         if [ x$rightfile = xy ]; then break; fi
 	# Some versions of bash appear to be buggy if the same
 	# $_lockfile is opened repeatedly. Close the current fd here.
