diff --git a/configure.ac b/configure.ac
index 5407a008..496d5728 100644
--- a/configure.ac
+++ b/configure.ac
@@ -306,7 +306,7 @@ dnl $2; if yes, executes $3, otherwise - $4.
 AC_DEFUN([RKT_IF_HAS_VALUE],
          [value_found=0
           RKT_ITERATE([$2],[value],
-                              [AS_IF([AS_ECHO([",$1,"]) | grep --silent ",${value},"],
+                              [AS_IF([AS_ECHO([",$1,"]) | grep -q ",${value},"],
                                      [value_found=1
                                       break])])
           AS_IF([test ${value_found} -eq 1],[:;$3],[:;$4])])
@@ -585,7 +585,7 @@ AS_IF([test "x${RKT_RUN_FUNCTIONAL_TESTS}" = 'xyes' -o "x${RKT_RUN_FUNCTIONAL_TE
                 dnl prerequisites for host flavor
                 [AS_VAR_IF([RKT_STAGE1_DEFAULT_FLAVOR], [host],
                            dnl checking if we are running systemd as init, -.mount is a mount unit for /.
-                           [AS_IF([systemctl 2>/dev/null | grep --silent -e '-\.mount'],
+                           [AS_IF([systemctl 2>/dev/null | grep -q -e '-\.mount'],
                                   dnl systemd runs as init
                                   [sdv=`systemctl --version | head -1 | sed -e 's/^systemd \(@<:@0-9@:>@*\)$/\1/'`
                                    AS_IF([test ${sdv} -lt 227],
diff --git a/makelib/misc.mk b/makelib/misc.mk
index c28cb35c..3af372ac 100644
--- a/makelib/misc.mk
+++ b/makelib/misc.mk
@@ -278,7 +278,7 @@ endef
 # 1 - from
 # 2 - to
 define bash-cond-rename
-if cmp --silent "$1" "$2"; then rm -f "$1"; else mv "$1" "$2"; fi
+if cmp -s "$1" "$2"; then rm -f "$1"; else mv "$1" "$2"; fi
 endef

 # Generates a rule for generating a depmk for a given go binary from a
diff --git a/stage1/prepare-app/prepare-app.c b/stage1/prepare-app/prepare-app.c
index a1bb7ab8..4782dd16 100644
--- a/stage1/prepare-app/prepare-app.c
+++ b/stage1/prepare-app/prepare-app.c
@@ -256,6 +256,7 @@ ptmx_device_usable (int dirfd, const char *path)
 	is_char = S_ISCHR (st.st_mode);
 	dev_type = (expected_dev == st.st_rdev);

+#define ACCESSPERMS 0777
 	perms = (st.st_mode & ACCESSPERMS);

 	world_readable = (perms & WORLD_READABLE) == WORLD_READABLE;
diff --git a/stage1/usr_from_coreos/build-usr.mk b/stage1/usr_from_coreos/build-usr.mk
index cf3c736b..4e1b6e28 100644
--- a/stage1/usr_from_coreos/build-usr.mk
+++ b/stage1/usr_from_coreos/build-usr.mk
@@ -164,7 +164,7 @@ $(call generate-stamp-rule,$(CBU_MKBASE_STAMP),$(CCN_SQUASHFS) $(CBU_COMPLETE_MA
 	CBU_SQROOT=$$$$(unsquashfs -ls "$(CCN_SQUASHFS)" --no-progress | tail --lines=1); \
 	unsquashfs -ls "$(CCN_SQUASHFS)" | grep "^$$$${CBU_SQROOT}" | sed -e "s/$$$${CBU_SQROOT}\///g" | sort >"$(CBU_SQUASHFS_FILES)"; \
 	comm -1 -2 "$(CBU_SQUASHFS_FILES)" "$(CBU_COMPLETE_MANIFEST)" >"$(CBU_COMMON_FILES)"; \
-	if ! cmp --silent "$(CBU_COMMON_FILES)" "$(CBU_COMPLETE_MANIFEST)"; \
+	if ! cmp -s "$(CBU_COMMON_FILES)" "$(CBU_COMPLETE_MANIFEST)"; \
 	then \
 		echo -e "Files listed in $(CBU_COMPLETE_MANIFEST) are missing from $(CCN_SQUASHFS):\n$$$$(comm -1 -3 "$(CBU_SQUASHFS_FILES)" "$(CBU_COMPLETE_MANIFEST)")"; \
 		exit 1; \
