diff --git a/tools/firmware/hvmloader/hvmloader.c b/tools/firmware/hvmloader/hvmloader.c
index 598a226..0c3963a 100644
--- a/tools/firmware/hvmloader/hvmloader.c
+++ b/tools/firmware/hvmloader/hvmloader.c
@@ -135,9 +135,11 @@ static void init_hypercalls(void)
 
         if ( !strcmp("XenVMMXenVMM", signature) )
             break;
+        if ( !strcmp("ZenZenZenZen", signature) )
+            break;
     }
 
-    BUG_ON(strcmp("XenVMMXenVMM", signature) || ((eax - base) < 2));
+    //BUG_ON( (strcmp("XenVMMXenVMM", signature) && strcmp("ZenZenZenZen", signature) ) || ((eax - base) < 2));
 
     /* Fill in hypercall transfer pages. */
     cpuid(base + 2, &eax, &ebx, &ecx, &edx);
diff --git a/tools/libxl/libxl_create.c b/tools/libxl/libxl_create.c
index 32d45dc..283b41b 100644
--- a/tools/libxl/libxl_create.c
+++ b/tools/libxl/libxl_create.c
@@ -320,6 +320,8 @@ int libxl__domain_build_info_setdefault(libxl__gc *gc,
         libxl_defbool_setdefault(&b_info->u.hvm.acpi_laptop_slate,  false);
         libxl_defbool_setdefault(&b_info->u.hvm.nx,                 true);
         libxl_defbool_setdefault(&b_info->u.hvm.viridian,           false);
+        libxl_defbool_setdefault(&b_info->u.hvm.spoof_viridian,     false);
+        libxl_defbool_setdefault(&b_info->u.hvm.spoof_xen,          false);
         libxl_defbool_setdefault(&b_info->u.hvm.hpet,               true);
         libxl_defbool_setdefault(&b_info->u.hvm.vpt_align,          true);
         libxl_defbool_setdefault(&b_info->u.hvm.altp2m,             false);
@@ -1545,6 +1547,10 @@ static void domcreate_launch_dm(libxl__egc *egc, libxl__multidev *multidev,
         console.backend_domid = state->console_domid;
         libxl__device_console_add(gc, domid, &console, state, &device);
         libxl__device_console_dispose(&console);
+        LOG(DEBUG, "Checking spoofing for guest (domid %d): xen %d, vir %d", domid,
+                                        libxl_defbool_val(d_config->b_info.u.hvm.spoof_xen),
+                                        libxl_defbool_val(d_config->b_info.u.hvm.spoof_viridian)
+                         );
 
         if (libxl_defbool_val(d_config->b_info.u.hvm.vkb_device)) {
             libxl_device_vkb_init(&vkb);
diff --git a/tools/libxl/libxl_x86.c b/tools/libxl/libxl_x86.c
index cdb294a..ed3ac19 100644
--- a/tools/libxl/libxl_x86.c
+++ b/tools/libxl/libxl_x86.c
@@ -414,6 +414,16 @@
             LOG(ERROR, "Couldn't set HVM_PARAM_VPT_ALIGN");
             goto out;
         }
+        if (xc_hvm_param_set(xch, domid, HVM_PARAM_SPOOF_XEN,
+                             libxl_defbool_val(info->u.hvm.spoof_xen))) {
+            LOG(ERROR, "Couldn't set HVM_PARAM_SPOOF_XEN");
+            goto out;
+        }
+        if (xc_hvm_param_set(xch, domid, HVM_PARAM_SPOOF_VIRIDIAN,
+                             libxl_defbool_val(info->u.hvm.spoof_viridian))) {
+            LOG(ERROR, "Couldn't set HVM_PARAM_SPOOF_VIRIDIAN");
+            goto out;
+        }
         if (info->u.hvm.mca_caps &&
             xc_hvm_param_set(CTX->xch, domid, HVM_PARAM_MCA_CAP,
                              info->u.hvm.mca_caps)) {
diff --git a/tools/libxl/libxl_types.idl b/tools/libxl/libxl_types.idl
index 7921950..246e128 100644
--- a/tools/libxl/libxl_types.idl
+++ b/tools/libxl/libxl_types.idl
@@ -569,6 +569,8 @@ libxl_domain_build_info = Struct("domain_build_info",[
                                        ("viridian",         libxl_defbool),
                                        ("viridian_enable",  libxl_bitmap),
                                        ("viridian_disable", libxl_bitmap),
+                                       ("spoof_viridian",   libxl_defbool),
+                                       ("spoof_xen",        libxl_defbool),
                                        ("timeoffset",       string),
                                        ("hpet",             libxl_defbool),
                                        ("vpt_align",        libxl_defbool),
diff --git a/tools/misc/xen-detect.c b/tools/misc/xen-detect.c
index fd187a4..7752219 100644
--- a/tools/misc/xen-detect.c
+++ b/tools/misc/xen-detect.c
@@ -83,6 +83,8 @@ static int check_for_xen(int pv_context)
 
         if ( !strcmp("XenVMMXenVMM", signature) && (regs[0] >= (base + 2)) )
             goto found;
+        if ( !strcmp("ZenZenZenZen", signature) && (regs[0] >= (base + 2)) )
+            goto found;
     }
 
     return 0;
diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index 4723f5d..311e8a5 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -4274,6 +4274,14 @@ static int hvmop_set_param(
     case HVM_PARAM_IOREQ_SERVER_PFN:
         d->arch.hvm.ioreq_gfn.base = a.value;
         break;
+    case HVM_PARAM_SPOOF_XEN:
+        printk("spoof_xen %" PRId64 " - curr dom %d, dom %d\n", a.value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_xen = a.value;
+        break;
+    case HVM_PARAM_SPOOF_VIRIDIAN:
+        printk("spoof_vir %" PRId64 " - curr dom %d, dom %d\n", a.value, curr_d->domain_id, d->domain_id);
+        d->arch.hvm.spoof_viridian = a.value;
+        break;
     case HVM_PARAM_NR_IOREQ_SERVER_PAGES:
     {
         unsigned int i;
diff --git a/xen/arch/x86/hvm/viridian/viridian.c b/xen/arch/x86/hvm/viridian/viridian.c
index 44c8e6c..b45bdbf 100644
--- a/xen/arch/x86/hvm/viridian/viridian.c
+++ b/xen/arch/x86/hvm/viridian/viridian.c
@@ -121,9 +121,13 @@ void cpuid_viridian_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = 0x40000006; /* Maximum leaf */
-        memcpy(&res->b, "Micr", 4);
-        memcpy(&res->c, "osof", 4);
-        memcpy(&res->d, "t Hv", 4);
+	printk("wetware cpuid_viridian_leaves spoof_vir\n"); // "Wetware Labs"
+        //memcpy(&res->b, "Micr", 4);
+        //memcpy(&res->c, "osof", 4);
+        //memcpy(&res->d, "t Hv", 4);
+        memcpy(&res->b, "Wetw", 4);
+        memcpy(&res->c, "era ", 4);
+        memcpy(&res->d, "Labs", 4);
         break;
 
     case 1:
diff --git a/xen/arch/x86/traps.c b/xen/arch/x86/traps.c
index 97499a0..b500095 100644
--- a/xen/arch/x86/traps.c
+++ b/xen/arch/x86/traps.c
@@ -892,9 +892,19 @@ void cpuid_hypervisor_leaves(const struct vcpu *v, uint32_t leaf,
     {
     case 0:
         res->a = base + limit; /* Largest leaf */
-        res->b = XEN_CPUID_SIGNATURE_EBX;
-        res->c = XEN_CPUID_SIGNATURE_ECX;
-        res->d = XEN_CPUID_SIGNATURE_EDX;
+        if (!d->arch.hvm.spoof_xen ) {
+            printk("cpuid_hypervisor_leaves - real id. domid %d\n",d->domain_id);
+               res->b = XEN_CPUID_SIGNATURE_EBX;
+               res->c = XEN_CPUID_SIGNATURE_ECX;
+               res->d = XEN_CPUID_SIGNATURE_EDX;
+        } else
+        {
+            printk("cpuid_hypervisor_leaves - spoofed id. domid %d\n",d->domain_id);
+               res->b = ZEN_CPUID_SIGNATURE_EBX;
+               res->c = ZEN_CPUID_SIGNATURE_ECX;
+               res->d = ZEN_CPUID_SIGNATURE_EDX;
+        }
+
         break;
 
     case 1:
diff --git a/xen/include/asm-x86/hvm/domain.h b/xen/include/asm-x86/hvm/domain.h
index bcc5621..a506710 100644
--- a/xen/include/asm-x86/hvm/domain.h
+++ b/xen/include/asm-x86/hvm/domain.h
@@ -159,6 +159,8 @@ struct hvm_domain {
     bool_t                 mem_sharing_enabled;
     bool_t                 qemu_mapcache_invalidate;
     bool_t                 is_s3_suspended;
+    bool_t                 spoof_xen;
+    bool_t                 spoof_viridian;
 
     /*
      * TSC value that VCPUs use to calculate their tsc_offset value.
diff --git a/xen/include/public/arch-x86/cpuid.h b/xen/include/public/arch-x86/cpuid.h
index ce46305..7e63c55 100644
--- a/xen/include/public/arch-x86/cpuid.h
+++ b/xen/include/public/arch-x86/cpuid.h
@@ -53,6 +53,10 @@
 #define XEN_CPUID_SIGNATURE_ECX 0x65584d4d /* "MMXe" */
 #define XEN_CPUID_SIGNATURE_EDX 0x4d4d566e /* "nVMM" */
 
+#define ZEN_CPUID_SIGNATURE_EBX 0x5A6e655A /* "ZenZ" */
+#define ZEN_CPUID_SIGNATURE_ECX 0x655A6e65 /* "enZe" */
+#define ZEN_CPUID_SIGNATURE_EDX 0x6e655A6e /* "nZen" */
+
 /*
  * Leaf 2 (0x40000x01)
  * EAX[31:16]: Xen major version.
diff --git a/xen/include/public/hvm/params.h b/xen/include/public/hvm/params.h
index 36832e4..2be4178 100644
--- a/xen/include/public/hvm/params.h
+++ b/xen/include/public/hvm/params.h
@@ -302,7 +302,10 @@
 #define HVM_PARAM_MCA_CAP 38
 #define XEN_HVM_MCA_CAP_LMCE   (xen_mk_ullong(1) << 0)
 #define XEN_HVM_MCA_CAP_MASK   XEN_HVM_MCA_CAP_LMCE
+#define HVM_PARAM_SPOOF_XEN    39
 
-#define HVM_NR_PARAMS 39
+#define HVM_PARAM_SPOOF_VIRIDIAN    40
+
+#define HVM_NR_PARAMS          41
 
 #endif /* __XEN_PUBLIC_HVM_PARAMS_H__ */
diff --git a/tools/xl/xl_parse.c b/tools/xl/xl_parse.c
index 36832e4..2be4178 100644
--- a/tools/xl/xl_parse.c
+++ b/tools/xl/xl_parse.c
@@ -1670,6 +1670,10 @@
         xlu_cfg_get_defbool(config, "vpt_align", &b_info->u.hvm.vpt_align, 0);
         xlu_cfg_get_defbool(config, "apic", &b_info->apic, 0);
 
+        /* For spoofing Xen */
+        xlu_cfg_get_defbool(config, "spoof_xen", &b_info->u.hvm.spoof_xen, 0);
+        xlu_cfg_get_defbool(config, "spoof_viridian", &b_info->u.hvm.spoof_viridian, 0);
+
         switch (xlu_cfg_get_list(config, "viridian",
                                  &viridian, &num_viridian, 1))
         {
